apps/testing-router-outlet/src/app/app.component.spec.ts

```diff
@@ -1,31 +1,152 @@
+
+      // There is only 1 button in the html
+      const button = screen.getByRole('button');
+      expect(button).toBeDisabled();
     });
   });
 
   describe('Given a search criteria with no book match', () => {
     it('Then shows No book found', async () => {
-      //todo
+      await setup();
+
+      const input = screen.getByRole('textbox');
+      const button = screen.getByRole('button');
+
+      expect(input).toBeInTheDocument();
+
+      await userEvent.type(input, 'asdfasdfasdf');
+
+      expect(
+        screen.queryByText('Search criteria is required!')
+      ).not.toBeInTheDocument();
+
+      await userEvent.click(button);
+
+      expect(
+        screen.queryByText('No book found for this search')
+      ).toBeInTheDocument();
+
+      // Error: Not implemented: navigation (except hash changes)
     });
   });
 
   describe('Given a search criteria with one book match', () => {
     it('Then shows One book and no error', async () => {
-      //todo
+      await setup();
+
+      const input = screen.getByRole('textbox');
+      const button = screen.getByRole('button');
+
+      expect(input).toBeInTheDocument();
+
+      await userEvent.type(input, '1984');
+
+      expect(
+        screen.queryByText('Search criteria is required!')
+      ).not.toBeInTheDocument();
+
+      await userEvent.click(button);
+
+      expect(
+        screen.queryByText('Borrowed Book: 1984 by George Orwell')
+      ).toBeInTheDocument();
     });
   });
 
+  // 100% coverage before last 2 tests
+
   describe('Given a search criteria in Uppercase with one book match', () => {
     it('Then shows One book and no error', async () => {
-      //todo
+      await setup();
+
+      const input = screen.getByRole('textbox');
+      const button = screen.getByRole('button');
+
+      expect(input).toBeInTheDocument();
+
+      await userEvent.type(input, 'ANIMAL FARM');
+
+      expect(
+        screen.queryByText('Search criteria is required!')
+      ).not.toBeInTheDocument();
+
+      await userEvent.click(button);
+
+      expect(
+        screen.queryByText('Borrowed Book: Animal Farm by George Orwell')
+      ).toBeInTheDocument();
     });
   });
 
   describe('Given a search criteria with multple books matches', () => {
     it('Then shows a list of books', async () => {
-      //todo
+      await setup();
+
+      const input = screen.getByRole('textbox');
+      const button = screen.getByRole('button');
+
+      expect(input).toBeInTheDocument();
+
+      await userEvent.type(input, 'Tolkien');
+
+      expect(
+        screen.queryByText('Search criteria is required!')
+      ).not.toBeInTheDocument();
+
+      await userEvent.click(button);
+
+      expect(
+        screen.queryByText(
+          'Borrowed Book: The Lord of the Rings by J.R.R. Tolkien'
+        )
+      ).toBeInTheDocument();
+      expect(
+        screen.queryByText('Borrowed Book: The Hobbit by J.R.R. Tolkien')
+      ).toBeInTheDocument();
```

Comment:
be careful, you need to create a new git branch and start from main branch again. Those commit should not be here.